src/main.js

```diff
@@ -13,4 +13,19 @@ export const enableGeoTIFFTileSource = (OpenSeadragon) => {
   OpenSeadragon.GeoTIFFTileSource = GeoTIFFTileSource;
 };
 
+// Run an IIFE to attach the GeoTIFFTileSource to the OpenSeadragon namespace
+// IF OpenSeadragon is available in the global scope
+(function (global, factory) {
+  // Skip if currently in ESM mode
+  if (typeof exports === "undefined") {
+    return;
+  }
```

Comment:
I'm not totally familiar with what this check will do. If you say it is doing the right thing I think we can merge...